Neos.Neos/Classes/Fusion/Helper/CachingHelper.php

```diff
@@ -70,15 +71,46 @@ public function nodeTag($nodes)
     /**
      * Generate an `@cache` entry tag for a node type
      * A cache entry with this tag will be flushed whenever a node
-     * (for any variant) that is of the given node type (including inheritance)
-     * is updated.
+     * (for any variant) that is of the given node type(s)
+     * (including inheritance) is updated.
      *
-     * @param NodeType $nodeType
+     * @param string|NodeType|string[]|NodeType[] $nodeType
+     * @return string[]
+     */
+    public function nodeTypeTag($nodeType) : array
+    {
+        if (!is_array($nodeType) && !($nodeType instanceof \Traversable)) {
+            $result = [$this->getNodeTypeTagFor($nodeType)];
+            return array_filter($result);
+        }
+
+        $result = [];
+        foreach ($nodeType as $singleNodeType) {
+            $result[] = $this->getNodeTypeTagFor($singleNodeType);
+        }
+
+        return array_filter($result);
+    }
+
+    /**
+     * @param $nodeType
      * @return string
      */
-    public function nodeTypeTag(NodeType $nodeType)
+    protected function getNodeTypeTagFor($nodeType)
     {
-        return 'NodeType_' . $nodeType->getName();
+        $nodeTypeName = '';
+        if (is_string($nodeType)) {
+            $nodeTypeName = $nodeType;
+        }
+        if ($nodeType instanceof NodeType) {
+            $nodeTypeName = $nodeType->getName();
+        }
+
+        if ($nodeTypeName === '') {
+            return '';
```

Comment:
See together with comment above, I wanted to ensure that this ALWAYS returns a string no matter what.